Core/DataAccess/EntityFramework/EfEntityRepositoryBase.cs

```diff
@@ -0,0 +1,59 @@
+﻿using Microsoft.EntityFrameworkCore;
+using System.Linq.Expressions;
+using WebNetSample.Core.Entities;
+
+namespace Core.DataAccess.EntityFramework
+{
+    public class EfEntityRepositoryBase<TEntity, TContext> : IEntityRepository<TEntity>
```

Comment:
For part2: Actually that's why i'm using `EF` to specify which ORM I'am using to make it more understandable for people. When devs want to use Dapper in the future, they can name it like `DapperEntityRepositoryBase`. Otherwise, it might cause a prolem that people can't understand which ORM they're using but if you are saying that's wrong, I can still make it just `BaseRepository`

Reply:
Yep, you are absolutely right, but, in my opinion, this approach makes sense when you are sure that there will be another ORM. In our case it would just make code too specific, but ok, let's leave it as it is.